mne/encoding/feature.py

```diff
@@ -0,0 +1,157 @@
+                             " found shape %s" % event_ixs.shape)
+        event_ids = np.ones_like(event_ixs) if event_ids is None else event_ids
+        unique_event_types = np.unique(event_ids)
+        if event_dict is None:
+            event_dict = dict((ii, 'event_%s' % ii)
+                              for ii in unique_event_types)
+
+        # Turn event_ixs from seconds to indices in case sfreq != 1
+        event_ixs = event_ixs * self.sfreq
+        event_ixs = event_ixs.astype(int)
+
+        # Iterate through event types and create columns of event onsets.
+        events_continuous = np.zeros([len(unique_event_types), self.n_times])
+        event_names = []
+        for ii, ev_type in enumerate(unique_event_types):
+            msk_events = event_ids == ev_type
+            i_ev = event_ixs[msk_events]
+            events_continuous[ii, i_ev] = 1
+
+            # Handle event names
+            event_names.append(event_dict[ev_type])
+        self.event_names = event_names
+        self.events_continuous = events_continuous
+        self.unique_event_types = unique_event_types
+
+    def transform(self, X=None, y=None):
+        """Return continuous events data.
+
+        Returns
+        -------
+        events_continuous : array, shape (n_unique_events, n_times)
+            A binary array with one row for each unique event type. 1s
+            in the array correspond to event onsets for that row.
+        """
+        return self.events_continuous
+
+    def fit_transform(self, event_ixs, event_ids=None, event_dict=None):
+        self.fit(event_ixs, event_ids=event_ids, event_dict=event_dict)
+        return self.transform()
+
+
+class DataSubsetter(object):
+    """Return a subset of data."""
+    def __init__(self, ixs):
+        self.ixs = ixs.astype(int)
+
+    def fit(self, data, y=None):
+        if data.shape[-1] < self.ixs.max():
+            raise ValueError('ixs exceed data shape')
+        self.data = np.asarray(data)
+
+    def transform(self, X=None, y=None):
+        return self.data[..., self.ixs]
+
+    def fit_transform(self, ixs, y=None):
+        self.fit(ixs)
+        return self.transform()
+
+
+def delay_timeseries(ts, sfreq, delays):
+    """Return a time-lagged input timeseries.
+
+    Parameters
+    ----------
+    ts: array, shape (n_feats, n_times)
+        The timeseries to delay
+    sfreq: int
+        The sampling frequency of the series
+    delays: list of floats
+        The time (in seconds) of each delay. Negative means
+        timepoints in the past, positive means timepoints in
+        the future.
+
+    Returns
+    -------
+    delayed: array, shape(n_feats * n_delays, n_times)
+        The delayed matrix
+    """
+    delayed = np.zeros([len(delays), ts.shape[-1]])
+    for ii, delay in enumerate(delays):
+        roll_amount = -1 * int(delay * sfreq)
+        rolled = np.roll(ts, roll_amount, axis=-1)
+        if roll_amount < 0:
+            rolled[:, roll_amount:0] = 0
```

Comment:
Hey - going through these comments now. I think this should be OK. The reasoning is that if roll_amount is negative, then doing `rolled[:, roll_amount:]` will go `roll_amount` *backwards* from index 0 up to index 0. The reason being that is `np.roll` gets a negative value, then it'll take the first N indices from the beginning of the vector and put them as the last N indices of the returned vector. We basically just want to turn those to 0.

E.g.:

`a = np.arange(1, 6)`
`b = np.roll(a, -2)`  
gives `[3, 4, 5, 1, 2]`.

We want the 1 and the 2 to be reset to zero, so then we do

`b[-2:] = 0`

and now we have `[3, 4, 5, 0, 0]`. Which is what we want, no?